content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -0,0 +1,1387 @@
+
+#### Advanced
+- **Evaluate context** is used for optimization in the graph when using successive iterable nodes. If this property is set to true, each iteration in this node will also calculate all looped context feeding this node. If there are several iterable nodes connected in sequence, the calculations become exponential with every iteration of the current node calculating every iteration of the previous node back to the beginning of the chain. This can cause a spike in process time in the graph. Disabling evaluate context will prevent nodes from initiating evaluations of previous context each iteration which will greatly speed up the graph. However, since there may be a case where it is desireable to re-evaluate context with each iteration of a node, this parameter is available to the user and defaults to true. If a graph seems to be taking too long to process, check iterable nodes to see if disabling the evaluation of context will help improve performance.
+
+#### Inputs
+- **geometry0** can optionally be connected to a node providing a Geometry type.
+- **geometry1** can optionally be connected to a node providing a Geometry type.
+- **geometry2** can optionally be connected to a node providing a Geometry type.
+- **geometry3** can optionally be connected to a node providing a Geometry type.
+- **geometry4** can optionally be connected to a node providing a Geometry type.
+- **geometry5** can optionally be connected to a node providing a Geometry type.
+- **geometry6** can optionally be connected to a node providing a Geometry type.
+- **geometry7** can optionally be connected to a node providing a Geometry type.
+- **geometry8** can optionally be connected to a node providing a Geometry type.
+- **geometry9** can optionally be connected to a node providing a Geometry type.
+
+#### Outputs
+- **output** is a randomly selected geometry from those available through connectioned inputs.
+
+![Compute normals node](/img/tools/nge/computeNormalsNode.jpg)
+
+### ComputeNormals
+There are times when it is necessary to recompute normals when creating geometry. This node is simply there to handle any computations of normals that are needed. 
+
+#### Advanced
+- **Mapping**
+
+#### Inputs
+- **geometry** is connected to a node providing a Geometry type.
+
+#### Outputs
+- **output** is the geometry after computing normals.
+
+![Mapping node](/img/tools/nge/mappingNode.jpg)
+
+### Mapping
+There are times when it is necessary to recompute normals when creating geometry. This node is simply there to handle any computations of normals that are needed. Simply pass the vertices to be mapped as well as the center of the projection to create UV mappings for each vertex.

```

Comment:
As I don't know what the `Mapping` block is doing I may be wrong, but it seems suspicious that the first two sentences are the same than for the `ComputeNormals` block(?)

Reply:
Thank you for catching this one! This is a UV mapping node. I had meant to change "recompute normals" to "project UV coordinates" and forgot. Likely something distracted me as I was writing this paragraph and I had to step away and forgot where I was when I came back.

Reply:
fixed in https://github.com/BabylonJS/Documentation/pull/963/commits/5395b92d4e3c0f28c82faf008026b09b521801ce